dist/array_extensions.js

```diff
@@ -55,4 +55,12 @@
 		return this.map(callback);
 	}
 
+	function take(size, callback) {
+		if (typeof size !== 'number') {
+			throw new TypeError('Excepted a number');
+		}
+
+		return this.filter(callback).slice(0, size);
```

Comment:
This is not the most performant solution, it will iterate a huge list of elements, apply the criteria and then *take* the desired elements.